rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -184,8 +227,48 @@ int COCOFileSourceReader::release()
 
 void COCOFileSourceReader::reset()
 {
-    if (_shuffle)
+    _aspect_ratios.clear();
+
+    if (_meta_data_reader && _meta_data_reader->aspect_ratio_grouping()) {
+        for (const auto &filename : _file_names) {
+            std::string base_filename = filename.substr(filename.find_last_of("/\\") + 1);
+            auto img_size = _meta_data_reader->lookup_image_size(base_filename);
+            auto aspect_ratio = static_cast<float>(img_size.h) / img_size.w;
+            _aspect_ratios.push_back(aspect_ratio);
+        };
+
+        // zip the two vectors
+        std::vector<std::pair<std::string, float>> zipped(_file_names.size());
+        for (size_t i = 0; i < _file_names.size(); i++)
+            zipped[i] = std::make_pair(_file_names[i], _aspect_ratios[i]);
+
+        // sort according to aspect ratios
+        std::sort(zipped.begin(), zipped.end(), [](auto &lop, auto &rop)
+                    { return lop.second < rop.second; });
+
+        // extract sorted file_names
+        std::vector<std::string> sorted;
+        std::transform(zipped.begin(), zipped.end(), std::back_inserter(sorted), [](auto &pair)
+                        { return pair.first; });
+
```

Comment:
since most of the operations seems same, can we create one function as common and call from initialize and reset?

Reply:
Optimized the AR grouping and removed the repeated operations